packages/main/src/plugin/kubernetes-client.ts

```diff
@@ -1049,64 +1010,159 @@ export class KubernetesClient {
+    const manifests = await this.loadManifestsFromFile(filePath);
+    await this.syncResources(context, manifests, 'create', namespace);
+  }
+
+  /**
+   * Create Kubernetes resources on the specified cluster. Resources are created sequentially.
    *
    * @param context the context name to use
    * @param manifests the list of Kubernetes resources to create
+   * @param namespace the namespace to use for any resources that don't include one
    */
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  async createResources(context: string, manifests: any[], optionalNamespace?: string): Promise<void> {
-    let telemetryOptions = {};
+  async createResources(context: string, manifests: any[], namespace?: string): Promise<void> {
+    await this.syncResources(context, manifests, 'create', namespace);
+  }
+
+  /**
+   * Apply a given yaml file to a context, i.e. 'kubectl apply -f'. Resources that exist
+   * on the context are patched, and any new resources are created.
+   *
+   * @param context a context
+   * @param filePath file system path to a YAML Kubernetes spec
+   * @param namespace the namespace to use for any resources that don't include one
+   * @return an array of resources created
+   */
+  async applyResourcesFromFile(context: string, filePath: string, namespace?: string): Promise<KubernetesObject[]> {
+    const manifests = await this.loadManifestsFromFile(filePath);
+    return this.syncResources(context, manifests, 'apply', namespace);
+  }
+
+  /**
+   * Apply a given yaml file to a context, i.e. 'kubectl apply -f'. Resources that exist
+   * on the context are patched, and any new resources are created.
+   *
+   * @param context a context
+   * @param filePath file system path to a YAML Kubernetes spec
+   * @param namespace the namespace to use for any resources that don't include one
+   * @return an array of resources created
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  async applyResources(context: string, manifests: any[], namespace?: string): Promise<KubernetesObject[]> {
+    return this.syncResources(context, manifests, 'apply', namespace);
+  }
+
+  /**
+   * Applies or creates a set of resources to a context, via creation or patching.
+   *
+   * @param context a context
+   * @param manifests a set of Kubernetes spec manifests
+   * @param action 'create' (only create new resources, do not overwrite existing) or 'apply' (create or patch as necessary)
+   * @param namespace the namespace to use for any resources that don't include one
+   * @return an array of resources created
+   *
+   * Heavily influenced by the API example:
+   * https://github.com/kubernetes-client/javascript/blob/0fbfd8fc2dcc7f4ec3e6fcd64a5c55169b6ef0b8/examples/typescript/apply/apply-example.ts
+   */
+
+  async syncResources(
+    context: string,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    manifests: any[],
```

Comment:
avoid the any usage, use what is used inside the function like  `{kind?: unknown, metadata?:.....}[]`

Reply:
Changed this (in sync and apply).